pkg/webhook/server_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/internal/cert"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/internal/cert/generator"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/internal/cert/writer"
+	"sigs.k8s.io/testing_frameworks/integration/addr"
+)
+
+type fakeCertWriter struct {
+	changed bool
+}
+
+func (cw *fakeCertWriter) EnsureCert(dnsName string) (*generator.Artifacts, bool, error) {
+	return &generator.Artifacts{}, cw.changed, nil
+}
+
+func (cw *fakeCertWriter) Inject(objs ...runtime.Object) error {
+	return nil
+}
+
+var _ = Describe("webhook server", func() {
+	Describe("run", func() {
+		var stop chan struct{}
+		var s *Server
+		var cn = "example.com"
+
+		BeforeEach(func() {
+			port, _, err := addr.Suggest()
+			Expect(err).NotTo(HaveOccurred())
+			s = &Server{
+				sMux: http.NewServeMux(),
+				ServerOptions: ServerOptions{
+					Port: int32(port),
+					BootstrapOptions: &BootstrapOptions{
+						Host: &cn,
+					},
+				},
+			}
+
+			cg := &generator.SelfSignedCertGenerator{}
+			s.CertDir, err = ioutil.TempDir("/tmp", "controller-runtime-")
```

Comment:
Are we doing cleanup of this temp dir ?

Reply:
Good catch.
Added code to cleanup the temp dir.
https://github.com/kubernetes-sigs/controller-runtime/pull/260/files#diff-8b0412ec4fd52af1419bd0fcd0f2e101R159